### ISSUE TITLE
No resource, method, or local variable named `slave_jar_resource' when creating windows jenkins slave

### ISSUE THREAD

**Issue by bingnet** [sous-chefs/jenkins#665]
### Cookbook version
5.0.5

### Chef-client version
Chef 12.19.36

### Platform Details
windows-2012r2

### Scenario:
When converging/setting-up a windows slave vm, the jenkins slave_windows.rb resource seems to not find slave_jar_resource, and gets the following error:
`No resource, method, or local variable named `slave_jar_resource' for `Chef::Provider::JenkinsWindowsSlave ""'`

NOTE:  When I comment-out lines calling slave_jar_resource in the slave_windows.rb, the slave gets created.

### Steps to Reproduce:
execute a jenkins windows slave block

```
              jenkins_windows_slave 'Builder' do
                name "Builder_#{node.name}"
                winsw_url node['jenkins']['slave']['windows']['winsw_url']
                winsw_checksum node['jenkins']['slave']['windows']['winsw_checksum']
                user node['jenkins']['slave']['windows']['service']['username']
                password node['jenkins']['slave']['windows']['service']['password']
                remote_fs node['jenkins']['slave']['windows']['remote_fs']
                labels node['jenkins']['slave']['labels']
                description node['jenkins']['slave']['description']
                executors node['jenkins']['slave']['executors']
              end
```

### Expected Result:
The windows slave converges and gets created.

### Actual Result:
```
NameError
           ---------
           No resource, method, or local variable named `slave_jar_resource' for `Chef::Provider::JenkinsWindowsSlave ""'
```

**Comment by sdhoka** [comment]
Facing similar issue

**Comment by skylerto** [comment]
I am also having this issue.

**Comment by skylerto** [comment]
Looks like there is some work being done on this. Ref issue: #670

**Comment by scotthain** [comment]
@tas50 @skylerto @bingnet I believe this is resolved in 6.0.0 - I just tested my infra and it is working.

**Comment by skylerto** [comment]
@scotthain you're right, this should be closed. #672 was merged.
